Documentation/Upgrade/rook-upgrade.md

```diff
@@ -50,18 +51,24 @@ those releases.
     official releases. Builds from the master branch can have functionality changed or removed at any
     time without compatibility support and without prior notice.
 
-## Breaking changes in v1.16
+## Breaking changes in v1.17
 
-* The minimum supported Kubernetes version is v1.27.
+* The minimum supported Kubernetes version is v1.27
```

Comment:
Looks like K8s 1.33 is due out in a couple weeks, I'm thinking we should increase the min version to v1.28 now with our 1.17 release. @subhamkrai Want to update the CI and docs for a new min version?

Reply:
1.30 is the oldest version with upstream support. My production env is in the process of upgrading from 1.30 -> 1.31. Interestingly, it looks like gke still allows 1.27: https://cloud.google.com/kubernetes-engine/docs/release-notes#current_versions

Reply:
For Rook, just to be clear it's more a [statement](https://rook.io/docs/rook/latest-release/Getting-Started/release-cycle/#k8s-versions) of where we will run our CI to have some validation against different K8s versions, even if the versions are out of official support cycle.

Reply:
Changed to 1.28

Reply:
@travisn https://github.com/rook/rook/pull/15670 also @BlaineEXE I'm updating the version in my PR as well let me know if you want me to revert or you will just to avoid conflicts. Thanks